File: 205/py/205.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        dictS = {}
        dictT = {}

        n = len(s)

        for i in range(n) :
            chrS = s[i]
            chrT = t[i]

            if dictS.get(chrS):
                if dictS[chrS] != chrT :
                    return False
            else :
                dictS[chrS] = chrT

            if dictT.get(chrT):
                if dictT[chrT] != chrS :
                    return False
            else :
                dictT[chrT] = chrS

        return True

File: 205/go/205.go
func isIsomorphic(s string, t string) bool {
    mapS := make(map[byte]byte)
    mapT := make(map[byte]byte)

    n := len(s)

    for i := 0; i < n; i++ {
        chrS := s[i]
        chrT := t[i]
        valS, findS := mapS[chrS]

        if findS {
            if valS != chrT {
                return false
            }
        }else{
            mapS[chrS] = chrT
        }

        valT, findT := mapT[chrT]

        if findT {
            if valT != chrS {
                return false
            }
        }else{
            mapT[chrT] = chrS
        }
    }

    return true
}